The Forest Expansion

Murderfoxen opened the pull request:
- Adds an entire area to the left-hand side of the map (Accessed by the Latejoin Carriage Entrance)

- Moves the druid's grove to the forest, making them the only playerspawn in the area.

- Replaces their old base with a swamp.

- adds a handful of fun encounters in the area for players to explore and discover.

- Adds some cinnabar deposits for mages who have struggled to find cinnabar when mining into the cliffside of the forest.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request
- Adds an entire area to the left-hand side of the map (Accessed by the Latejoin Carriage Entrance)

- Moves the druid's grove to the forest, making them the only playerspawn in the area.

- Replaces their old base with a swamp.

- adds a handful of fun encounters in the area for players to explore and discover.

- Adds some cinnabar deposits for mages who have struggled to find cinnabar when mining into the cliffside of the forest.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->



## Why It's Good For The Game

it adds further content to the map. The dungeons can be used to assist in the summoning process for wizards as well as some things to do for players who are seeking to go out in groups. 

- adds a second mana fountain for magic casters game cycle
- adds black obelisks that people can fight
- adds cinnabar spawns which are very low currently so having dedicated deposits will add to the magic casters game cycle
- adds new mineral deposits for people who wish to mine the forest cliffside
- adds some fae mobs as minor fights and encounters to those who travel too deep
- adds some elemental mobs as a primitive dungeon with nearby ores to mine
- gives additional options towards alternative playstyles; like those who might want somewhere to build a forested hut, etc.
- lots of wood to chop and places to hunt wolves.
- It has no access from antagonists immediately and isn't as directly lethal as the bog so it presents itself as a safer option when in reality it has different hazards to the bog for players to face

I could add ways for the bandits/goblins to access this forest as well; making it more challenging and inclusive of players from both sides. I feel it would need the addition of a vanguard outpost to make sense with the city if this were included.

![Screenie6](https://github.com/user-attachments/assets/3b69493b-3145-4802-9726-48d4ef781a2f)
![Screenie5](https://github.com/user-attachments/assets/c5d0f3fc-22dc-4475-aa5d-c96aa3ed440c)
![Screenie4](https://github.com/user-attachments/assets/b7ca55de-1453-455d-b220-ccae26fbf9fb)
![Screenie3](https://github.com/user-attachments/assets/67d7a180-a2e7-46b0-ad0b-6c19adbb7f53)
![screenie1](https://github.com/user-attachments/assets/a9edd7ba-f7e2-4f38-8f8a-1a01deeabc63)
![screenie2](https://github.com/user-attachments/assets/74ec42d2-7b48-4097-b0a0-dec9a324796e)

- It clears up the area near the east for what could be an expansion/revision of the harbor to be exterior of High-Town or something else from fellow mappers. (This is something I'd like to also work on but requires remodeling of the sewers below which the goblins utilize, potentially moving the cemetary and a lot of other technical aspects)

![bogreplace](https://github.com/user-attachments/assets/0dd1ab64-0342-48a1-81de-e0453e689761)

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Proof of Testing (Required)

It's a map, I haven't /tested/ it, but I'm also a perfectionist. I'll be suprised if I made any mistakes that are game-breaking from my experience doing this in the past.

<!-- Show proof of testing, screenshots or recordings for features, proof of compile for backend changes -->

Murderfoxen commented:
Mages will still need to visit the bog to gain volcanic shards, whilst the bog remains the only place for leylines as well. Given that bandits and liches exist I think spellcasters will travel to either location and still face perils. 🤷‍♀️

RadiantFlash7 commented:
Right. I actually Rather like the idea of making use of the space to the west. At some point, I'd prefer if we added a second entrance to town, likely on the west side, to reduce it's 'secureness' because It's a little... overly secure at the moment. More ways for nare-do-wells to enter town is probably good. But thats a seperate PR from what you've done.

The cinnabar deposits are fine, moving the druids is- fine I suppose.

The things I do take issue with however: There should not be a 'safe' mana fountain. Mages should need to go to the bog for it, which is why it's all the way in the south of the bog.
Black obelisks are intentionally rare as their lootdrop can be used to make null rings, or summon a mob that can probably single-handedly wipe out the town. (Admittedly, this has never been done in game before, Because it's /very/ difficult. But still.)
My other issue at present, is the summon mobs being placed on the map.
All summon mobs have a 'timer' at which when they see no human mobs around or don't have a target, they despawn. Partly as an anti-cheese measure.
So having them on round start will just- have them disappear leaving behind loot used to summon them.

RadiantFlash7 commented:
All those points aside, we have a fairly major map change test merged at present, so this will need to wait once the adjustments are made to it.

JohnnyJohnJohnson commented:
This would be awesome, maybe add one or two Towner homes and a lake nearby if these aren't already there.

NikolayByondPlayer commented:
Would be better to make another entrance/exit to the main area not to make it too isolated
Probably the current forest area near the fortress can also be cut down a bit not to make too much empty spaces.
Some random generations for big zones could also improve the place by making it less stale.

But yeah looks nice overall except for that weird Г-movement you will always need to be doing to get there - more entraces could be good, maybe the place could be moved more south too

Murderfoxen commented:
I'm waiting for a time/confirmation on when I could work on this before picking it back up.
I don't know when is a good time to bother as it is a large enough PR I really need to be given a set and fixed working time to transfer what I have done here to the current version of the map. It's a big space to work on and people could add their map changes to this but I think it would take priority and requires more feedback before being merged. 


 As of sending this I haven't bothered to edit it or the mobs yet; though I think the summon code should be refactored so they don't disappear.  These mobs would be fun for the community to fight and engage with, and would add to more than just the mages gameplay loop if they could be fought or treated as mobs/bosses accordingly. 

**Things I plan to change from RadiantFlashes comments:**
- Removing the dense obelisks entirely in favor of keeping it to the bog-specific side; as they do seem to fit the themes better in that location. 
 - Put the mana fountain deeper underground or in a dangerous spot to make it challenging to access, like the bog. (if this isnt sufficient I can remove it, but I do think it needs playtests and feedback to really determine how easy it existing makes the mage gameplay loop, though it is admittedly not as deep in the woods as I could change it to make it more challenging and similiarly as hard to reach as the bog location.)

**I'm also willing to:** 

- Expose the western side of the city to the forest to expand on this.
- Add some lodges to the pathside section, small cabins which are primarily abandoned or fallen into ruin as 'fixer-uppers' that add some flavor to the border of the city
- make the forest further-west look more 'corrupted' if those themes benefit the lore of ratwood; giving a sense of darker woods or some encroaching darkness/rot of a spiritual nature
- Make another entrance/exit from the bog that leads to this area so antagonists can spawn here.
- Make travelgates for the bandits (also editting the bandit zone map with a new pathway from the north that leads to the forest) to give them a new place to attack and inhabit.
- Update the spawn list for the forest to include whatever mobs are considered appropriate for the forest; such as wolves, rats, spiders and other beasts. I would like to exclude skeletons however for thematic reasons, instead increasing the amount of wolves or other mobs that spawn here to balance it out. 
- Add summon mobs as ambush encounter mobs, as this -would- work in accordance to the proximity timer code and add a variety of supernatural creatures to make it unique and potentially even more dangerous than the bog in some aspects. (Fairies/fey in the woods as rarer ambush spawns)

Murderfoxen commented:
> Having mage stuff in here kind of defeats the purpose of making mages go to the bog in the first place

I feel that having them go to the bog isn't necessary to their gameplay loop. Having played mages for a while - going alone is a death-sentence for almost every person new to the role; yet having the fountain there is useful for antagonists such as bandits, vampire servants and lich servants. I think both can exist with balancing and making sure it doesn't become too safe of a zone or stagnant to danger from players and mobs.

Murderfoxen commented:
> Would be better to make another entrance/exit to the main area not to make it too isolated Probably the current forest area near the fortress can also be cut down a bit not to make too much empty spaces. Some random generations for big zones could also improve the place by making it less stale.
> 
> But yeah looks nice overall except for that weird Г-movement you will always need to be doing to get there - more entraces could be good, maybe the place could be moved more south too

I agree; and the idea of a gatehouse from lowtowns entrance would make the area feel more secure - but add to players feeling split or stressed to defend the city. I chose to give it a natural pathway with no gates because it was connected to lowtown, but A river with a bridge similiar to the bog would be nice. 

I'll look into adding some random generation too; though I'm not really familiar with the limits of it - If a forest zone has any generation and I zone it; it would add trees to spaced areas and block off those sections and make changing paths. I can also add log/tree or bush/tree spawners that more reliably change the map in controlled ways.

Murderfoxen commented:
> This would be awesome, maybe add one or two Towner homes and a lake nearby if these aren't already there.

I think it would be best if I left them as 'abandoned' shacks to emphasise that this area is not safe; letting players build cabins or repair them. This is definately something somebody else can change later though. Let's see how it goes.

Murderfoxen commented:
![Whole Area changed](https://github.com/user-attachments/assets/5cf13994-2b1a-4743-98ab-f2511d01adab)
Before I work on the map, this gives a good idea for those who arent opening the map themselves and checking it of the scale. 
Please -do- open the map and finecomb anything not listed above; I plan on working on it if I get the greenlights I need. 

I haven't heard any feedback from druid players despite this moving their spawn really far away (aside from latejoins) so I'd really like to hear about how they feel. I mainly play druid/mage so I'm a bit biased here and would appreciate other voices.

Murderfoxen commented:
![image](https://github.com/user-attachments/assets/28b241af-37d2-441a-b3fa-b7d42a24d8f5)
Changed the obelisk area to be an abandoned hut near the edge of the map. Good spot for antags or what have you nae'er'do'wells as an outpost. 

![image](https://github.com/user-attachments/assets/84a84c30-ee9e-469c-ac4e-03b540b504ca)
Moved the mana fountain to be much harder to access. This area now functions as a cave that links to the druids grove from the path. 

![image](https://github.com/user-attachments/assets/ae0e9fe2-7a34-4272-8000-d6e3f7b9855e)
The mana fountain is now arguably harder than the bogs to access despite the travel time being shorter. You have to gather your adventuring party, because these mobs are not to be trifled with. (Requires variables that allow mobs to be persistent, or new variants that are entirely different which are to be created.)

![image](https://github.com/user-attachments/assets/14d37bb0-e75c-4cca-8336-f4058011175a)
 Small foresters hut added near the entrance, features a drying rack and a mediocre bed with hearth. 
 
 **Important features:**

![image](https://github.com/user-attachments/assets/7c94ff70-d490-466c-bf5a-dc2c16410cec)

Opened up the west-side wall to face the waterside of high-town. Finally, those walls at the keep to the west can actually view something! (if it werent so -dark- and -wet- down there!)

- this allows for antagonists, as well as druids or whoever wishes to to quickly access the sewer-side and just behind the tavern if they really wanted to. I'd still take the road if I were them.

Murderfoxen commented:
![image](https://github.com/user-attachments/assets/25fe04c1-3a96-46ac-917d-f6498b3faf71)
added two waterfalls that add a sense of flow, added smoke to simulate mist. Not actually sure if it works, so I'm going to have to test it by compile/running now. These areas use /river  zones to add atmospheric flowing water. Really adds to the nature vibes.

Genevesh commented:
From the way you described it here or on discord, I really dig the idea, but if you're planning to make static mobs or ambushes, I would rather they had FAR fewer drops, or no drops at all, since it somewhat defeats the purpose of summoning, with how easy these dungeons will be to cheese with ranged spells.

Other than that, I'm all hands for this, I think summon mob ambushes specifically will be really awesome.
